rasa/core/policies/ted_policy.py

```diff
@@ -661,75 +998,374 @@ def _create_all_labels_embed(self) -> Tuple[tf.Tensor, tf.Tensor]:
 
         return all_label_ids, all_labels_embed
 
-    def _emebed_dialogue(
-        self, dialogue_in: tf.Tensor, sequence_lengths: tf.Tensor
-    ) -> Tuple[tf.Tensor, tf.Tensor]:
+    def _embed_dialogue(
+        self,
+        dialogue_in: tf.Tensor,
+        tf_batch_data: Dict[Text, Dict[Text, List[tf.Tensor]]],
+    ) -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor]:
         """Create dialogue level embedding and mask."""
-
-        mask = self._compute_mask(sequence_lengths)
+        dialogue_lengths = tf.cast(tf_batch_data[DIALOGUE][LENGTH][0], tf.int32)
+        mask = self._compute_mask(dialogue_lengths)
 
         dialogue_transformed = self._tf_layers[f"transformer.{DIALOGUE}"](
             dialogue_in, 1 - mask, self._training
         )
         dialogue_transformed = tfa.activations.gelu(dialogue_transformed)
 
-        if self.max_history_tracker_featurizer_used:
+        if self.use_only_last_dialogue_turns:
             # pick last vector if max history featurizer is used
```

Comment:
Don't think this is a valid comment as `use_only_last_dialogue_turns` is always set to `True` in inference mode.
```suggestion
```